src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopePatternGenerator.cpp

```diff
@@ -0,0 +1,91 @@
+
+
+#include <OpenMS/CHEMISTRY/Element.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopePatternGenerator.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+
+#include <cmath>
+#include <fstream>
+
+using namespace std;
+
+namespace OpenMS
+{
+  IsotopePatternGenerator::IsotopePatternGenerator() : IsotopeDistribution()
+  {}
+
+  IsotopePatternGenerator::IsotopePatternGenerator(const IsotopeDistribution& rhs) :
+    IsotopeDistribution(rhs)
+  {
+    
+  }
+  void IsotopePatternGenerator::merge(ContainerType& raw, double resolution)
+  {
+    //raw must be ordered to work correctly ascending order on power field
+
+    UInt output_size = ceil((raw.back().getMZ() - raw.front().getMZ())/resolution);
+    LOG_INFO << "output size " << output_size << endl;
+    LOG_INFO << "raw size " << raw.size() <<endl;
+
+    distribution_.clear();
+    distribution_.resize(output_size, Peak1D(0, 0));
+
+    for(auto& p : raw)
+    {
+      // Is this the case?
+      
+      UInt index = round((p.getMZ() - raw.front().getMZ())/resolution);
+      if(index >= distribution_.size()){
+
+        LOG_INFO << index <<endl;
+        
+      }
+      auto& mass = distribution_[index].getPosition() ;
+
+      mass = mass == 0 ?
+             raw.front().getMZ() * index :
+             mass;
+      distribution_[index].setIntensity(distribution_[index].getIntensity() + p.getIntensity());
+    }
+  }
+  
+  void IsotopePatternGenerator::setEmpiricalFormula(const EmpiricalFormula& formula)
+  {
+    formula_ = formula;
+  }
+
+
+
+  void IsotopePatternGenerator::dumpIDToFile(String file)
+  {
+    ofstream out(file.c_str());
+    for(auto& sample : distribution_)
+    {
+      out << sample.getMZ() << " "<<sample.getIntensity() << endl;
+    }
+    
+    out.close();
+  }
+  
+  /* Start of the midas interface */
+  MIDAs::MIDAs(EmpiricalFormula& formula, double resolution, UInt N_):
```

Comment:
`const EmpiricalFormula&` (here it's even more important, in current state you can't pass rvalues to the constructor)